src/lib/homestore_backend/index_kv.cpp

```diff
@@ -70,7 +71,59 @@ HSHomeObject::get_from_index_table(shared< BlobIndexTable > index_table, shard_i
         return folly::makeUnexpected(BlobError::UNKNOWN_BLOB);
     }
 
-    return index_value.pbas();
+    const auto& pbas = index_value.pbas();
+
+    // blob get API
+    if (state == BlobState::ALIVE) {
+        if (pbas == tombstone_pbas) { return folly::makeUnexpected(BlobError::TOMBSTONE_BLOB); }
+        return pbas;
+    }
+
+    // get tombstone, only used by test for now, can be used by GC thread if needed
+    if (state == BlobState::TOMBSTONE) {
+        if (!(pbas == tombstone_pbas)) { return folly::makeUnexpected(BlobError::ALIVE); }
+    }
+
+    // used for GC thread to range scan index table
```

Comment:
This comment is not needed, and we use shard_id to do range-query, not pba, right?

Reply:
yea, we use shard_id to do range-query, but we need to check the pba to determine whether this bolb has been deleted, since we put a tombstone pba in value

Reply:
No, we don't need to do that for GC, GC will call range-update to remove tombstoned key/values.